### PR TITLE
Add interactive setup for GitBark.

### PR DESCRIPTION
**Pull request by elibon99**

### CODE REVIEW

File: gitbark/commands/setup.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+        return
+
+    if not has_branch(project, branch):
+        cmd("git", "checkout", "--orphan", branch)
+        cmd("git", "reset", "--hard")
+    else:
+        cmd("git", "checkout", branch)
+
+
+def get_commit_rules(project: Project):
```
Comment (dainnilsson) [inline gitbark/commands/setup.py:100]: Maybe an invalid yaml file should still cause an error. A missing file can be replaced with {} I think, and then anything reading "rules" should use `.get("rules", [])` to allow the key to be missing. So if you have an error in your yaml, it doesn't just treat the file as missing.

File: gitbark/commands/setup.py [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+        file=f"{project.path}/{BARK_IN_REPO_FOLDER}/{COMMIT_RULES}",
+        content=yaml.safe_dump(commit_rules, sort_keys=False),
+    )
+
+
+def add_branches_interactive(project: Project, branch: str) -> None:
+    curr_branch = cmd("git", "symbolic-ref", "--short", "HEAD")[0]
+    if curr_branch != BARK_RULES_BRANCH:
+        checkout_or_orphan(project, BARK_RULES_BRANCH)
+    try:
```
Comment (dainnilsson) [inline gitbark/commands/setup.py:162]: If there is an error in bark_rules we should probably fail instead. Same for other calls to get_bark_rules that catch Exception

**Review by dainnilsson — CHANGES_REQUESTED** [review]

File: gitbark/commands/setup.py (hunk 1 above)
Comment (elibon99) [reply gitbark/commands/setup.py]: Pushed a commit that resolves this.

File: gitbark/commands/setup.py (hunk 2 above)
Comment (elibon99) [reply gitbark/commands/setup.py:224]: Pushed a commit that resolves this.

**Review by dainnilsson — APPROVED** [review]

**Pull request merged**